test/unit/media/headset.test.ts

```diff
@@ -0,0 +1,159 @@
+        headsetEvents$ = headsetLibrary.headsetEvents$;
+        sdk = new SimpleMockSdk() as any;
+        sdkHeadset = new SdkHeadset(sdk);
+    })
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    })
+
+    describe('constructor()', () => {
+        it('should start initialization', () => {
+            // const headset = new SdkHeadset(sdkMedia);
+            expect(sdkHeadset['sdk']).toBe(sdk);
+            expect(sdkHeadset['headsetLibrary']).toBe(headsetLibrary);
+            expect(sdkHeadset['headsetEvents$']).toStrictEqual(headsetEvents$);
+        })
+    })
+
+    describe('updateAudioInputDevice', () => {
+        it('should fetch the proper device and send it to the headset library', () => {
+            const testId = "testId";
+            const findCachedDeviceByIdAndKindSpy = jest.spyOn(sdk.media, 'findCachedDeviceByIdAndKind' as any).mockReturnValue({
+                kind: 'audioinput',
+                deviceId: 'testId',
+                label: 'Test Device Mark V',
+            } as MediaDeviceInfo);
+            const activeMicChangeSpy = jest.spyOn(headsetLibrary, 'activeMicChange' as any);
+            sdkHeadset.updateAudioInputDevice(testId);
+            expect(findCachedDeviceByIdAndKindSpy).toHaveBeenCalledWith(testId, 'audioinput');
+            expect(activeMicChangeSpy).toHaveBeenCalledWith('test device mark v');
+        })
+    })
+
+    describe('getCurrentSelectedImplementation()', () => {
+        it('should fetch the currently selected vendor implementation from the headset library', () => {
+            headsetLibrary.activeMicChange('plantronics test device');
+            expect(headsetLibrary.selectedImplementation).toStrictEqual(headsetLibrary['plantronics']);
+        })
+    })
+
+    describe('showRetry', () => {
+        it('should return false if the selected implementation has disableRetry as true', () => {
+            headsetLibrary.activeMicChange('plantronics test device');
+            headsetLibrary.selectedImplementation.disableRetry = true;
+            const showRetryResult = sdkHeadset.showRetry();
+            expect(showRetryResult).toBe(false);
+        })
+
+        it('should return true if disableRetry is false, isConnected is false and isConnecting is false', () => {
+            headsetLibrary.activeMicChange('plantronics test device');
+            headsetLibrary.selectedImplementation.disableRetry = false;
+            headsetLibrary.selectedImplementation.isConnecting = false;
+            const showRetryResult = sdkHeadset.showRetry();
+            expect(showRetryResult).toBe(true);
+        })
+
+        it('should return false if disableRetry is false, isConnected is false but isConnecting is true', () => {
+            headsetLibrary.activeMicChange('plantronics test device');
+            headsetLibrary.selectedImplementation.isConnecting = true;
+            let showRetryResult = sdkHeadset.showRetry();
```

Comment:
Codacy has a fix for the issue: ['showRetryResult' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
            const showRetryResult = sdkHeadset.showRetry();
```